chain/evm/provider/transactor_generator.go

```diff
@@ -0,0 +1,73 @@
+package provider
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// TransactorGenerator is an interface for generating bind transactor instances
+type TransactorGenerator interface {
+	// Generate creates a new Solana keypair and returns the private key
+	Generate(chainID *big.Int) (*bind.TransactOpts, error)
+}
+
+var (
+	_ TransactorGenerator = (*transactorFromRaw)(nil)
+	_ TransactorGenerator = (*transactorRandom)(nil)
+)
+
+// TransactorFromRaw creates a new instance of the privateKeyFromRaw generator with the raw private
+// key.
+func TransactorFromRaw(privKey string) TransactorGenerator {
+	return &transactorFromRaw{
+		PrivKey: privKey,
+	}
+}
+
+// transactorFromRaw is an EVM keypair generator that creates an keypair from a private key.
+type transactorFromRaw struct {
+	PrivKey string
```

Comment:
i think PrivKey can be private here
```suggestion
	privKey string
```